internal/baseservice/base_service.go

```diff
@@ -48,14 +41,6 @@ type Archetype struct {
 	TimeNowUTC func() time.Time
 }
 
-// WithSleepDisabled disables sleep in services that are using `BaseService`'s
-// `CancellableSleep` functions and returns the archetype for convenience. Use
-// of this is only appropriate in tests.
-func (a *Archetype) WithSleepDisabled() *Archetype {
```

Comment:
There are quite a few LOCs changed in this diff, but most of them are just removing this `WithSleepDisabled` feature from the base service. Having run into trouble with it multiple times now, I'd say it's just too dangerous/unintuitive. Services that need to disable sleep can add their own local flags for it.

Reply:
It does seem like a bit of a step back to spread out the problematic behavior into more places, but whatever you think is the best option here.

Reply:
Yeah ... I  think being able to disable sleep is good, but being able to _broadly_ disable it is just too dangerous/easy to miss. I'm totally fine trying to come up with a different/better generalized approach if we can find one, but more precise sleep disables for the relatively few test cases that need it seems like a reasonable compromise for now.